source/globals.d.ts

```diff
@@ -1,8 +1,7 @@
 /* eslint-disable @typescript-eslint/consistent-indexed-object-style */
 
 type AnyObject = Record<string, any>;
-// TODO: Fix this type (there's no such `delegate` global type) and enable the @typescript-eslint/no-redundant-type-constituents rule
```

Comment:
`@typescript-eslint/no-redundant-type-constituents` also complains about the `IntrinsicElements` declarations:

```
  source/globals.d.ts:41:21
  ✖  41:21  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents
  ✖  42:21  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents
  ✖  43:19  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents
  ✖  46:23  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents
  ✖  47:12  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents
  ✖  48:20  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents
  ✖  50:15  any overrides all other types in this intersection type.  @typescript-eslint/no-redundant-type-constituents

  7 errors
```

Reply:
Crap. Must have been some React types update.

Reply:
Still worth having this rule imo, I never even realized the `Deinit` was basically useless! Let's just disable it around this block until we figure out a fix.

Reply:
@fregante I'll finish this up and merge it tomorrow (Monday). What do you think about ↑